### ISSUE TITLE
SSH changes cause failure on retry

### ISSUE THREAD

**Issue by acozine** [curationexperts/ansible-hydra#141]
Steps to reproduce:
On CentOS 7, run a playbook that includes the ssh changes in the app-config role:

```
TASK [app-config : update ssh config] ******************************************
changed: [xxxxxx]

RUNNING HANDLER [app-config : restart ssh service] *****************************
changed: [xxxxxx]
```

Then re-run the playbook on the same host. You'll see:

```
TASK [setup] *******************************************************************
fatal: [xxxxxx]: UNREACHABLE! => {"changed": false, "msg": "SSH Error: data could not be sent to the remote host. Make sure this host can be reached over ssh", "unreachable": true}
```

The machine will still be reachable by ordinary ssh. Restore /etc/ssh/sshd_config and the problem will go away. Need to track down which setting is interfering with ansible here.

**Comment by acozine** [comment]
I discovered the issue - Ansible uses sftp to transfer files, and the path for the sftp executable differs from Ubuntu/Debian (`/usr/lib/openssh/sftp-server`) to CentOS (`/usr/libexec/openssh/sftp-server`). The sshd_config we're adding overwrote the correct path on CentOS. Fix is a change to the template, commit 0129cd9b6f607cc3334dde1b37721e49a39492c1.

**Comment by acozine** [comment]
Closed when Centos branch got merged in.
